core/iwasm/libraries/wasi-nn/src/wasi_nn.c

```diff
@@ -308,15 +327,17 @@ wasi_nn_compute(wasm_exec_env_t exec_env, graph_execution_context ctx)
     NN_DBG_PRINTF("Running wasi_nn_compute [ctx=%d]...", ctx);
 
     wasm_module_inst_t instance = wasm_runtime_get_module_inst(exec_env);
-    bh_assert(instance);
+    if (!instance) {
+        return runtime_error;
+    }
+
     WASINNContext *wasi_nn_ctx = wasm_runtime_get_wasi_nn_ctx(instance);
 
     wasi_nn_error res;
     if (success != (res = is_model_initialized(wasi_nn_ctx)))
         return res;
 
-    res = lookup[wasi_nn_ctx->current_encoding].compute(
-        wasi_nn_ctx->backend_ctx, ctx);
+    res = lookup.compute(wasi_nn_ctx->backend_ctx, ctx);
```

Comment:
All calls to lookup should be checked, right?

```suggestion
    if (lookup.compute) 
      res = lookup.compute(wasi_nn_ctx->backend_ctx, ctx);
    else
      NN_ERR_PRINTF("Backend not loaded.");
```

Reply:
Of course. Totally agree.